packages/core/src/components/text-field/text-field.tsx

```diff
@@ -203,7 +211,15 @@ export class TdsTextField {
           <div class="text-field-bar"></div>
 
           {usesSuffixSlot && (
-            <div class={`text-field-slot-wrap-suffix text-field-${this.state}`}>
+            <div
+              class={{
+                'text-field-slot-wrap-suffix': true,
+                'text-field-error': this.state === 'error',
+                'text-field-success': this.state === 'success',
+                'text-field-default': this.state === 'default',
+                'tds-u-display-none': this.readOnly,
```

Comment:
This is the big change right? 

Are the class changes above for some other reason?

Reply:
Could add this behaviour as part of the description for the `readOnly` and `slot="suffix"`

Reply:
@mJarsater  - Yes, when adding new class I realized we use string literal so I updated it to use object like structure that works better in realtime updates. Since I had to change it for lines you selected I felt it makes sense to update rest of the code. I wrote about it in description too.

Reply:
> Could add this behaviour as part of the description for the `readOnly` and `slot="suffix"`

Good idea, makes sense so we do not confuse users.